openwisp-monitoring/files/monitoring.agent

```diff
@@ -149,46 +176,36 @@ send_data() {
 			while true; do
 				if [ "$failures" -eq "$MAX_RETRIES" ]; then
 					[ -f "$RESPONSE_FILE" ] && error_message="\"$(cat "$RESPONSE_FILE")\"" || error_message='"".'
-					if [ "$VERBOSE_MODE" -eq "1" ]; then
-						logger -s "Data not sent successfully. Response code is \"$response_code\"." \
-							"Error message is $error_message" \
-							-p daemon.err
-					elif [ "$FAILING" -eq "0" ]; then
+					log -e -v "Data not sent successfully. Response code is \"$response_code\"." \
+						"Error message is $error_message"
+					# check if agent was already passing or not to avoid repeating log messages
+					if [ "$FAILING" -eq "0" ]; then
 						FAILING=1
-						logger -s "Data not sent successfully. Response code is \"$response_code\"." \
-							"Error message is $error_message" \
-							"Run with verbose mode to find more." \
-							-t openwisp-monitoring \
-							-p daemon.err
+						[ "$VERBOSE_MODE" -ne "1" ] && log -e -n "Data not sent successfully. Response code is \"$response_code\"." \
+							"Run with verbose mode to find more."
```

Comment:
There's still repetition here which makes the code hard to read.
I think we need to edit the log function to avoid this.

Ideally we could do something like:

```shell
log -e -n "Data not sent successfully. Response code is \"$response_code\"." \
    -v "Error message is $error_message"
```

The non verbose output should be printed also in verbose mode, while the verbose output is appended to the non verbose output but only when in verbose mode, so in non verbose mode we would have:

```
Data not sent successfully. Response code is "400".`
```

While in verbose mode we would have:

```
Data not sent successfully. Response code is "400". Error message is ERROR_MESSAGE_HERE.
```

Reply:
The condition is here to ensure that the logs are not added unnecessarily when the connection is interrupted due to some issue for a long time.

Discussed here in "Non-verbose mode" https://github.com/openwisp/openwrt-openwisp-monitoring/pull/42#pullrequestreview-708960736